inst/report/scripts/run_molevolvr_pipeline.R

```diff
@@ -0,0 +1,1296 @@
+    # Fetch data using epost and efetch
+    response <- GET(sprintf("https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=protein&format=xml&id=%s",
+                            acc))
+
+    # Check if the request was successful
+    if (response$status_code == 200) {
+      # Parse the XML response
+      parsed_xml <- read_xml(content(response, "text"))
+
+      # Extract relevant information
+      doc_summary <- xml_find_all(parsed_xml, ".//DocumentSummary")
+
+      for (summary in doc_summary) {
+        results <- rbind(results, data.frame(
+          Caption = xml_text(xml_find_first(summary, ".//Caption")),
+          Extra = xml_text(xml_find_first(summary, ".//Extra")),
+          Title = xml_text(xml_find_first(summary, ".//Title")),
+          Length = xml_text(xml_find_first(summary, ".//Slen")),
+          TaxID = xml_text(xml_find_first(summary, ".//TaxId")),
+          Organism = xml_text(xml_find_first(summary, ".//Organism")),
+          SourceDB = xml_text(xml_find_first(summary, ".//SourceDb")),
+          Completeness = xml_text(xml_find_first(summary, ".//Completeness")),
+          stringsAsFactors = FALSE
+        ))
+      }
+    } else {
+      warning(sprintf("Failed to fetch data for accession %s: %s",
+                      acc, response$status_code))
+    }
+  }
+
+  # Write results to the output file
+  write.table(results, file = outfile, sep = "\t", row.names = FALSE,
+              quote = FALSE)
+}
+
+# Main function to run based on the prefix
+run_acc2info <- function(infile, prefix, outdir) {
+  if (prefix == "NA") {
+    acc2info_phylo(infile, outdir)
+  } else {
+    acc2info(infile, prefix, outdir)
+  }
+}
+
+substitute_accnum_for_acc2info <- function(df_acc2info, df_header_map) {
+  df_result <- df_header_map |>
+    # set column name in header map to match accnum col in acc2info
+    dplyr::rename(AccNum = header_accnum) |>
+    # join onto header map
+    dplyr::left_join(df_acc2info, by = "AccNum") |>
+    # deselect accnum
+    dplyr::select(-AccNum) |>
+    # set the accnum col to the cleaned form
+    dplyr::rename(AccNum = header_clean) |>
+    # rm excess columns from header map file
+    dplyr::select(-header_original)
+  return(df_result)
+}
+
+
+replace_accession_numbers <- function(path_acc2info,
+                                      path_query_header_map, path_out) {
+
+  # Read the input files
+  df_acc2info <- read_tsv(path_acc2info)
+  df_query_header_map <- read_tsv(path_query_header_map)
+
+  # Substitute accession numbers
+  df_acc2info_substituted <- substitute_accnum_for_acc2info(df_acc2info,
+                                                            df_query_header_map)
+
+  # Print the substituted dataframe
+  print("### df_acc2info_substituted")
+  print(df_acc2info_substituted)
+
+  # Write the substituted dataframe to the output file
+  write_tsv(df_acc2info_substituted, file = path_out, col_names = TRUE)
+}
+
+
+run_deltablast <- function(infile, prefix, outdir,
```

Comment:
other mentions need to be fixed, too.
```suggestion
runDELTABLAST <- function(infile, prefix, outdir,
```